### PR TITLE
add gradle recipe: RemoveSettingsPluginManagement

### PR DESCRIPTION
**Pull request by sullis**
## What's changed?

new recipe:  RemoveSettingsPluginManagement

This recipe removes "pluginManagement" from Gradle settings.gradle

## What's your motivation?

At work, I have hundreds of Gradle projects. I need to remove "pluginManagement" from all of my Gradle projects.

## Anything in particular you'd like reviewers to focus on?

n/a

## Anyone you would like to review specifically?

n/a

## Have you considered any alternatives or workarounds?

n/a

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: rewrite-gradle/src/main/java/org/openrewrite/gradle/plugins/RemoveSettingsPluginManagement.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public class RemoveSettingsPluginManagement extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Remove Gradle settings pluginManagement";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Remove Gradle settings pluginManagement from `settings.gradle(.kts)`.";
```
Comment (timtebeek) [inline rewrite-gradle/src/main/java/org/openrewrite/gradle/plugins/RemoveSettingsPluginManagement.java:41]:
```suggestion
        return "Remove Gradle settings pluginManagement from `settings.gradle`.";
```

File: rewrite-gradle/src/main/java/org/openrewrite/gradle/plugins/RemoveSettingsPluginManagement.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.tree.J;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class RemoveSettingsPluginManagement extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Remove Gradle settings pluginManagement";
```
Comment (timtebeek) [inline rewrite-gradle/src/main/java/org/openrewrite/gradle/plugins/RemoveSettingsPluginManagement.java:36]:
```suggestion
        return "Remove Gradle settings `pluginManagement`";
```

**Comment by sullis** [comment]
CI build in progress.

**Comment by shanman190** [comment]
@sullis, can you provide some insight on the deeper context that you're trying to solve for?

At first glance, the deeper context is not apparent and the recipe itself feels a bit heavy at the moment from a removal scope by itself alone.

**Comment by sullis** [comment]
> @sullis, can you provide some insight on the deeper context that you're trying to solve for?
> 
> At first glance, the deeper context is not apparent and the recipe itself feels a bit heavy at the moment from a removal scope by itself alone.

I just updated the PR description.

**Comment by shanman190** [comment]
@sullis, so there is a lot of configuration that can happen within the `pluginManagement` block, so a blanket remove the entire block feels a bit odd/one off to me coming at it from an OSS recipe catalog viewpoint.

Some of the things that do come to mind are maybe we are failing to cleanup a now empty block after having say removed the last plugin repository. In this case, I'd almost expect for the remove repository recipe to cleanup empty blocks that it's left behind.

With that said, we have seen users create their own internal recipe libraries where they have put similar recipes where they know within their strict context that they want to be destructive even though a more general purpose recipe can't really be described or arrived at. Maybe that would work better here?

**Comment by sambsnyd** [comment]
I share Shannon's questioning about how general-use this recipe is. It seems quite specific to your scenario.

In your situation you know that this change is broadly desirable, so it isn't necessary for the recipe itself to be any smarter than this. For most other gradle builds most of the time removing the plugin management is going to break the build. If this were a recipe which could tell when plugin management is unnecessary/redundant and remove it only in those cases I would be more enthusiastic. If you're not interested in pursuing that maybe this is a better fit for an internal recipe module rather than the OSS catalogue.

**Comment by sullis** [comment]
> If you're not interested in pursuing that maybe this is a better fit for an internal recipe module rather than the OSS catalogue.

Sounds reasonable.  I'll close this PR.  I am going to make this an internal recipe.

**Pull request closed without merging**